example/centos/centos-9-x86_64-ami.yaml

```diff
@@ -1,37 +1,136 @@
 otk.version: "1"
 
-otk.define.default:
-  default:
-    language: "en_US.UTF-8"
-
-otk.define.centos-9-x86_64-ami:
-  architecture: "x86_64"
-  version: "9"
-  kernel_opts_list:
-    - "rd.blacklist=nouveau"
-    - "nvme_core.io_timeout=4294967295"
-  sshd_password_authentication: true
-  modfication:
-    language: "en_US.UTF-8"
-    timezone: "UTC"
-  enabled_services:
-    - sshd
-    - NetworkManager
-    - nm-cloud-setup.service
-    - nm-cloud-setup.timer
-    - cloud-init
-    - cloud-init-local
-    - cloud-config
-    - cloud-final
-    - reboot.target
-    - tuned
-  # todo: should this go into centos-9-common.yaml ? and we customize via normal "otk.defines" ?
+otk.define:
+  kernel_commandline: "console=tty0 console=ttyS0,115200n8 net.ifnames=0 nvme_core.io_timeout=4294967295"
   filesystem:
-    # this will expand to "root: ...\nboot: \n"etc (i.e. a real file-systems) and "kernel_opts_list:
-    # [list_item1, list_item2, ..]"
-    # (i.e. define the partition table as a map of root/boot/boot_efi vars)
-    # *and* it needs to undstand modifications (e.g. adding extra parititions somehow, size contraints)
-    otk.external.osbuild.create_partition_table_defines:
-      total_size: 10G
+    modifications:
+      filename: "image.raw"
+otk.include: "common/gen-partition-table-x86_64.yaml"
 
-otk.include.centos-9-x86_64: "common/centos-9.yaml"
+otk.target.osbuild:
+  pipelines:
+    - otk.include: "pipeline/build.yaml"
+    - name: os
+      build: name:build
+      stages:
+        - otk.include: "fragment/kernel-cmdline.yaml"
+        - type: org.osbuild.rpm
+        - type: org.osbuild.fix-bls
```

Comment:
extracting: org.osbuild.fix-bls,org.osbuild.keymap, org.osbuild.timezone would work here but because it's in a list it will look quite awkward

Reply:
It won't be *that* awkward, I believe we already use `otk.op.join` in some places and I used it 'everywhere' in my branch just so fragments could define one or more stages :)